src/backend/gl/gl_common.c

```diff
@@ -1181,11 +1177,12 @@ bool gl_init(struct gl_data *gd, session_t *ps) {
 	// buffer are skipped anyways, this should have no impact on performance.
 	GLint viewport_dimensions[2];
 	glGetIntegerv(GL_MAX_VIEWPORT_DIMS, viewport_dimensions);
-	gd->vp_height = viewport_dimensions[0];
-	gd->vp_width = viewport_dimensions[1];
+	gd->vp_width = viewport_dimensions[0];
```

Comment:
Can we remove `vp_width` and `vp_height`?

Reply:
We need these for setting up the correct projection matrix in the blur-context (`gl_create_blur_context()`).

Reply:
We could just use GL_MAX_VIEWPORT_DIMS there too.

Reply:
Indeed, we could just query `GL_MAX_VIEWPORT_DIMS` again. I think keeping this value in the global backend state is more consistent, as we assume the set viewport dimensions are globally applicable as well.

Reply:
I think it's a bit weird to keep a copy of a constant.

Reply:
Feel free to change if this is cleaner to you; no objections from my side.
Highly unlikely that the driver will report different values for the second query. :smile: